### PR TITLE
Remove jQuery from the repo release form

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched to plain JavaScript
- Tested the repo release form functionality and it works as before

# Demo using JavaScript without jQuery
action

### CODE REVIEW

File: web_src/js/features/repo-release.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-    const uuid = $(this).data('uuid');
-    const id = $(this).data('id');
-    $(`input[name='attachment-del-${uuid}']`).attr('value', true);
-    hideElem($(`#attachment-${id}`));
+  document.addEventListener('click', (e) => {
+    if (e.target.matches('.remove-rel-attach')) {
+      const uuid = e.target.getAttribute('data-uuid');
+      const id = e.target.getAttribute('data-id');
+      document.querySelector(`input[name='attachment-del-${uuid}']`).value = true;
+      hideElem(`#attachment-${id}`);
```
Comment (delvh) [inline web_src/js/features/repo-release.js:10]: Didn't the logic change in this line?
We invoke `document.querySelectorAll` here, but I don't think that can handle ID notation…

Comment (yardenshoham) [reply web_src/js/features/repo-release.js:10]: It works...
action

Comment (delvh) [reply web_src/js/features/repo-release.js:10]: Ah, it supports any CSS selector.
True, then it must still work…

**Review by delvh — APPROVED** [review]

File: web_src/js/features/repo-release.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-  $(document).on('click', '.remove-rel-attach', function() {
-    const uuid = $(this).data('uuid');
-    const id = $(this).data('id');
-    $(`input[name='attachment-del-${uuid}']`).attr('value', true);
-    hideElem($(`#attachment-${id}`));
+  document.addEventListener('click', (e) => {
+    if (e.target.matches('.remove-rel-attach')) {
+      const uuid = e.target.getAttribute('data-uuid');
+      const id = e.target.getAttribute('data-id');
+      document.querySelector(`input[name='attachment-del-${uuid}']`).value = true;
```
Comment (wxiaoguang) [inline web_src/js/features/repo-release.js]:
```suggestion
      document.querySelector(`input[name='attachment-del-${uuid}']`).value = 'true';
```

`value` should be a string.

File: web_src/js/features/repo-release.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/features/repo-release.js:10]: > Ah, it supports any CSS selector. True, then it must still work…

The `showElem/hideElem` handles everything. I have designed that to make it works for most cases, including jQuery replacement ....


image

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**